haystack_experimental/dataclasses/tool.py

```diff
@@ -2,11 +2,13 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
+import inspect
 from dataclasses import asdict, dataclass
 from typing import Any, Callable, Dict
 
 from haystack.lazy_imports import LazyImport
 from haystack.utils import deserialize_callable, serialize_callable
+from pydantic import create_model
```

Comment:
Let's add this as a direct dependency to the package.

Reply:
done